src/main/java/net/dv8tion/jda/api/entities/MessageSticker.java

```
@@ -0,0 +1,206 @@
+        this.tags = tags;
+    }
+
+    @Override
+    public long getIdLong()
+    {
+        return id;
+    }
+
+    /**
+     * The name of the sticker.
+     *
+     * @return the name of the sticker
+     */
+    @Nonnull
+    public String getName()
+    {
+        return name;
+    }
+
+    /**
+     * The description of the sticker.
+     *
+     * @return the description of the sticker
+     */
+    @Nonnull
+    public String getDescription()
+    {
+        return description;
+    }
+
+    /**
+     * The ID of the pack the sticker is from.
+     *
+     * @return the ID of the pack the sticker is from
+     */
+    @Nonnull
+    public String getPackId()
+    {
+        return Long.toUnsignedString(getPackIdLong());
+    }
+
+    /**
+     * The ID of the pack the sticker is from.
+     *
+     * @return the ID of the pack the sticker is from
+     */
+    public long getPackIdLong()
+    {
+        return packId;
+    }
+
+    /**
+     * The Discord hash-id of the sticker. This represents the actual asset file in the CDN for the sticker.
+     * <br><b>The URL for fetching sticker assets is currently private.
+     *
+     * @return the Discord hash-id of the sticker
+     */
+    @Nonnull
+    public String getAssetHash()
+    {
+        return asset;
+    }
+
+    /**
+     * The Discord hash-id of the preview of the sticker. This represents the actual preview asset file in the CDN for the sticker.
+     * <br><b>The URL for fetching sticker assets is currently private.
+     *
+     * @return the Discord hash-id of the preview image of the sticker or {@code null} if the sticker has no preview image
+     */
+    @Nullable
+    public String getPreviewAssetHash()
+    {
+        return previewAsset;
+    }
+
+    /**
+     * The url of the sticker asset.
+     *
+     * @return the url of the sticker
+     */
+    @Nonnull
+    public String getAssetUrl()
+    {
+        return String.format(ASSET_URL, id, asset, formatType.getExtension());
```

We should probably throw if the format type is unknown

---

We should also make it clear that this asset is not necessarily an image here

---

Throw in the `getExtension` or in this method itself?

---

Probably in this method

---

i think it should rather be in `getExtension` as that now returns `null` which is not really the best design

---

I don't really care either way, but if `getExtension` throws then it needs to be documented on both getters.